sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -0,0 +1,218 @@
+package io.sentry;
+
+import io.sentry.protocol.Contexts;
+import io.sentry.protocol.SentryId;
+import io.sentry.util.Objects;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class SentryTransaction extends SentryBaseEvent implements ISpan {
+  /** The transaction name. */
+  private @Nullable String transaction;
+
+  /** The moment in time when span was started. */
+  private final @NotNull Date startTimestamp;
+
+  /** The moment in time when span has ended. */
+  private @Nullable Date timestamp;
+
+  /** A list of spans within this transaction. Can be empty. */
+  private final @NotNull List<Span> spans = new CopyOnWriteArrayList<>();
+  /**
+   * A hub this transaction is attached to. Marked as transient to be ignored during JSON
+   * serialization.
+   */
+  private @NotNull final transient IHub hub;
+
+  /** The {@code type} property is required in JSON payload sent to Sentry. */
+  @SuppressWarnings("UnusedVariable")
+  private @NotNull final String type = "transaction";
+
+  /** Creates transaction. */
+  SentryTransaction(final @NotNull String name) {
+    this(name, new SpanContext(), NoOpHub.getInstance());
+  }
+
+  SentryTransaction(final @NotNull TransactionContext transactionContext, final @NotNull IHub hub) {
+    this(transactionContext.getName(), transactionContext, hub);
+  }
+
+  /**
+   * Creates transaction with name and contexts.
+   *
+   * @param name - transaction name
+   * @param contexts - transaction contexts
+   */
+  SentryTransaction(
+      final @NotNull String name, final @NotNull SpanContext contexts, final @NotNull IHub hub) {
+    Objects.requireNonNull(contexts, "contexts are required");
+    this.transaction = Objects.requireNonNull(name, "name is required");
+    this.startTimestamp = DateUtils.getCurrentDateTime();
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    Contexts ctx = new Contexts();
+    ctx.setTrace(contexts);
```

Comment:
L: Probably me not understanding the API yet but it's odd that `setTrace` doesn't take a `trace` but a `SpanContext` instead. Should it be `setSpanContext`?

Reply:
Then it would be `Contexts#setSpanContext` while we have `Contexts#setApp` etc. I think it can be understood as "span context is a tracing information". Thats the naming I figured from the docs, but I have no objections if you decide to change it.

Reply:
Just a `L` in the LOGAF scale :) lets leave it